src/runge_kutta.jl

```diff
@@ -0,0 +1,509 @@
+                               0 0 0 0 0 34//105 9//35 9//35 9//280 9//280 0     41//840 41//840],
+                               [0,    2//27, 1//9, 1//6 , 5//12, 1//2 , 5//6 , 1//6 , 2//3 , 1//3 , 1 , 0, 1]
+                            )
+
+
+###########################
+# Runge-Kutta solvers
+# (Hairer & Wanner 1992 p.134, p.165-169)
+
+function make_consistent_types(fn, y0, tspan, btab)
+    # There are a few types involved which somehow need to be consistent:
+    #
+    # Et = eltype(tspan)
+    # Ey = eltype(y0)
+    # Ef = eltype(Tf)
+    #
+    # There are also the types of the containers, but they are not
+    # needed as `similar` is used to make containers.
+    # Tt = typeof(tspan)
+    # Ty = typeof(y0)              # note, this can be a scalar
+    # Tf = typeof(F(tspan(1),y0))  # note, this can be a scalar
+    #
+    # Returns
+    # - Et: eltype of time, needs to be a real "continuous" type, at
+    #       the moment a FloatingPoint
+    # - Ey: eltype of y
+    # - Ef: eltype of fn(t,y)
+    #   --> both of these are set to typeof(y0[1]/tspan[1])
+    # - btab: tableau with entries converted to Et
+
+    Ey = typeof(y0[1]/(tspan[end]-tspan[1]))
+    Ef = Ey # this could be changed if function type become available
+
+    Et = promote_type(eltype(tspan), Float16)
+    @assert Et<:Real
+    @assert isleaftype(Et) # maybe shouldn't error, only warn.
+
+    btab_ = convert(Et, btab)
+    return Et, Ey, Ef, btab_
+end
+
+######
+# Fixed step Runge-Kutta method
+# TODO: iterator method
+function oderk_fixed(fn, y0, tspan, btab::TableauRKExplicit)
+    # Non-arrays y0 treat as scalar
+    fn_ = (t, y) -> fn(t, y[1])
+    t,y = oderk_fixed(fn_, [y0], tspan, btab)
+    return t,y[:]
+end
+function oderk_fixed{N,S}(fn, y0::AbstractVector, tspan,
+                          btab_::TableauRKExplicit{N,S})
+    # TODO: instead of AbstractVector use a Holy-trait
+    Et, Ey, Ef, btab = make_consistent_types(fn, y0, tspan, btab_)
+    dof = length(y0)
+    tsteps = length(tspan)
+    ys = similar(y0, Ey, dof, tsteps)
+    ys[:,1] = y0
+    tspan = convert(Vector{Et}, tspan)
+    # work arrays:
+    ks = similar(y0, Ef, dof, S)
+    ytmp = similar(y0, Ey, dof)
+    for i=1:length(tspan)-1
+        dt = tspan[i+1]-tspan[i]
+        ys[:,i+1] = ys[:,i] # this also allocates a bit...
+        for s=1:S
+            for d=1:dof # loop as ytmp[:] = ys[:,i] allocates!
+                ytmp[d] = ys[d,i]
+            end
+            calc_next_k!(ks, ytmp, ytmp, s, fn, tspan[i], dt, dof, btab)
+            for d=1:dof
+                ys[d,i+1] += dt * btab.b[s]*ks[d,s]
+            end
+        end
+    end
+    return tspan, ys
+end
+
+# calculates k[s]
+function calc_next_k!{N,S}(ks::Matrix, ytmp::Vector, y, s, fn, t, dt, dof, btab::TableauRKExplicit{N,S})
+    # Calculates the next ks and puts it into ks[:,s]
+    # - ks and ytmp are modified inside this function.
+    ytmp[:] = y
```

Comment:
`ytmp[:] = y` is fast, at least for arrays (it uses copy! internally).  I tested this just today:
https://gist.github.com/mauro3/dfa1d24adaaa1bf48d7d

I'm aware that the scalar version is slower.  I haven't quite figured it out yet but I'll look into it.  One thing I noted that a lot of time was spent in `stepsize_hw92`, which actually needs some tweaking for scalars.